azure-kusto-ingest/azure/kusto/ingest/_ingest_client.py

```diff
@@ -48,63 +58,70 @@ def ingest_from_dataframe(self, df, ingestion_properties):
 
         fd = FileDescriptor(temp_file_path)
 
-        blob_name = "{db}__{table}__{guid}__{file}".format(
-            db=ingestion_properties.database, table=ingestion_properties.table, guid=uuid.uuid4(), file=file_name
-        )
-
-        containers = self._resource_manager.get_containers()
-        container_details = random.choice(containers)
-        storage_client = CloudStorageAccount(container_details.storage_account_name, sas_token=container_details.sas)
-        blob_service = storage_client.create_block_blob_service()
-
-        blob_service.create_blob_from_path(
-            container_name=container_details.object_name, blob_name=blob_name, file_path=temp_file_path
-        )
-
-        url = blob_service.make_blob_url(container_details.object_name, blob_name, sas_token=container_details.sas)
-
-        self.ingest_from_blob(BlobDescriptor(url, fd.size), ingestion_properties=ingestion_properties)
+        ingestion_properties.format = DataFormat.csv
+        self.ingest_from_file(fd, ingestion_properties)
 
         fd.delete_files()
         os.unlink(temp_file_path)
 
     def ingest_from_file(self, file_descriptor, ingestion_properties):
-        """Enqueuing an ingest command from local files.
+        """Ingest from local files.
         :param file_descriptor: a FileDescriptor to be ingested.
         :param azure.kusto.ingest.IngestionProperties ingestion_properties: Ingestion properties.
         """
-        file_descriptors = list()
-        containers = self._resource_manager.get_containers()
 
         if isinstance(file_descriptor, FileDescriptor):
             descriptor = file_descriptor
         else:
             descriptor = FileDescriptor(file_descriptor)
 
-        file_descriptors.append(descriptor)
-        blob_name = "{db}__{table}__{guid}__{file}".format(
-            db=ingestion_properties.database,
-            table=ingestion_properties.table,
-            guid=uuid.uuid4(),
-            file=descriptor.stream_name,
-        )
-
-        container_details = random.choice(containers)
-        storage_client = CloudStorageAccount(container_details.storage_account_name, sas_token=container_details.sas)
-        blob_service = storage_client.create_block_blob_service()
-
-        blob_service.create_blob_from_stream(
-            container_name=container_details.object_name, blob_name=blob_name, stream=descriptor.zipped_stream
-        )
-        url = blob_service.make_blob_url(container_details.object_name, blob_name, sas_token=container_details.sas)
-
-        self.ingest_from_blob(
-            BlobDescriptor(url, descriptor.size, descriptor.source_id), ingestion_properties=ingestion_properties
-        )
+        if self._use_streaming_ingest and descriptor.size < self._streaming_ingestion_size_limit:
+            if (
+                ingestion_properties.format == DataFormat.json
```

Comment:
same as above,
 `ingestion_properties.format in [DataFormat.json,  DataFormat.singlejson,  DataFormat.avro]`